charms/worker/k8s/lib/charms/k8s/v0/k8sd_api_manager.py

```diff
@@ -570,6 +570,17 @@ def update_cluster_config(self, config: UpdateClusterConfigRequest):
         body = config.dict(exclude_none=True)
         self._send_request(endpoint, "PUT", EmptyResponse, body)
 
+    def configure_dns(self, dns_domain: str, dns_ip: str):
+        """Configure the DNS for the k8s cluster.
+
+        Args:
+            dns_domain (str): The domain name for the DNS.
+            dns_ip (str): The IP address for the DNS.
+        """
+        endpoint = "/1.0/k8sd/cluster/config"
+        body = {"dns.dns-ip": dns_ip, "dns.cluster-domain": dns_domain, "dns.enabled": False}
+        self._send_request(endpoint, "POST", EmptyResponse, body)
+
```

Comment:
I don't think it's a good idea to create a specific method in API manager solely for configuring DNS. We aim to keep everything as clean as possible in the library. Could you refactor the logic in the charm code to directly call `update_cluster_config`, using the Pydantic models to achieve the same outcome?

Reply:
I will have to update it so that it takes the body as an argument to `update_cluster_config`

Reply:
Good catch thank you. This was dead code from my previous implementation.